mlir/lib/Dialect/Tosa/IR/TosaOps.cpp

```diff
@@ -503,6 +504,41 @@ LogicalResult tosa::ConcatOp::inferReturnTypeComponents(
   return success();
 }
 
+LogicalResult ConcatOp::verify() {
+  OperandRange inputs = getInput1();
+
+  auto inputRank = ShapedType::kDynamic;
+  bool hasRankedInputs;
```

Comment:
This value could be used uninitialized. Can we replace the loop by `llvm::all_of`
like
```
bool hasRankedInputs = llvm::all_of([]({return ...hasRank()};
if(!hasRankedInputs)
  return success();

// TODO: Is it valid for the concat to have zero inputs? 
auto inputRank = inputs[0].getRank();
 for (auto input : inputs) {
     ...
      if (inputRank != inputType.getRank()) {
        return emitOpError()
               << "rank of input " << inputType
               << " does not match other input rank(s) (" << inputRank << ")";
      }
    }
  }
```